lib/zaddress.js

```diff
@@ -59,7 +60,7 @@ function zSecretKeyToTransmissionKey(a_sk) {
   sk_enc[0] &= 248;
   sk_enc[32] &= 127;
   sk_enc[31] |= 64;
-  return Buffer.from(sodium.crypto_scalarmult_base(sk_enc)).toString('hex');
+  return Buffer.from(scalarMult.base(sk_enc)).toString('hex');
```

Comment:
Considering the risk of having an invalid public key let's extend unit tests with regression checks by verifying a batch of tuples of PK,SK pre-generated by zend

Reply:
@albertog78 test added  and passed with 500 addresses

Reply:
`zSpendingKey` and `zAddress` come directly from zend. `zSecretKey` derived from `zSpendingKey` using bs58check, `aPk` and `pkEnc` calculated from `zSecretKey` with zencashjs v1.2.3 using the libsodium wrappers.